### DISCUSSION TITLE
License for Qodana Image Expired

### DISCUSSION THREAD

**Discussion by esafirm** [JetBrains/Qodana#151] [Q&A] [answered]
Hi, we're using Qodana with Gitlab CI.
And we recently got this error message in our Qodana job

```
License of this Qodana image is expired. Please use "docker pull" to update image.
```

We can see from our pods event log that we're pulling the image. So we're not sure what to do.

How can we resolve this? Thanks

**Comment by brichbash** [comment, marked as answer]
Hello,

Could you please make sure that you're pulling the image with `2022.3-eap` or `latest` tag?

**Reply by esafirm** [reply]
@brichbash 

We currently use `2022.2-eap`

Does that mean we have to keep using `latest` and update our script accordingly in our pipeline?
It would be great if you can send a reference to this thing since it's quite confusing for me.
Thanks

**Reply by brichbash** [reply]
Yes, if you want to configure Qodana once and forget about it, you can use the tag `latest`. It will pull the current stable version of Qodana (right now it is `2022.3-eap`). We also publish less stable versions (for example, we're publishing `2023.1-eap` soon), but they can be accessed only using their tag.
However, you should be aware that if you're using the `latest` tag and our baseline feature, with the release of a new version, you might want to update your baseline, as a new major version can bring new checks (it is applicable if you use our `qodana.recommended` or `qodana.starter` profiles).

**Reply by esafirm** [reply]
> configure Qodana once and forget about it

This is not possible I believe since there are breaking changes in some versions. For example in 2022.3 the binary is changed from `/opt/idea/bin/entrypoint` to `qodana`

So there's no way to use a version and change it when we want to change it?
What's the reason behind this?

If there's no new version released or maybe delayed, how can we deal with it?

**Reply by brichbash** [reply]
> So there's no way to use a version and change it when we want to change it?

It is a limitation of our EAP for  linters. Once Qodana is released, this won't be the case. I'm sorry for the inconvenience. If you're using our jvm linter, then you can consider using [jvm-community](https://www.jetbrains.com/help/qodana/qodana-jvm-community.html) which doesn't have an expiring evaluation license, and you can stick to whatever version you want.

> If there's no new version released or maybe delayed, how can we deal with it?

We monitor license expiration dates for our linters and publish new images before them, so there shouldn't be such a situation. 
Even if for some reason a license is expired and there are no newer images. We'll be alerted and we'll publish new images as soon as possible (the next morning, if it happened at night).

**Reply by esafirm** [reply]
Thanks for your explanation.

> then you can consider using [jvm-community](https://www.jetbrains.com/help/qodana/qodana-jvm-community.html) which doesn't have an expiring evaluation license

We are currently using [Qodana Community for Android](https://www.jetbrains.com/help/qodana/qodana-jvm-android.html) that unfortunately has this 30 days license.

Is there any plan to remove the license since it's also a "community" edition?

**Reply by brichbash** [reply]
>... that unfortunately has this 30 days license.

Actually, it has a 60-day license.

> Is there any plan to remove the license since it's also a "community" edition?

We will do it once we consider the linter stable enough. Unfortunately, I can't provide you with any ETA.

**Reply by esafirm** [reply]
Thanks for all the answers, very helpful 👍
